packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts

```diff
@@ -417,6 +433,13 @@ export class TaskDefinition extends TaskDefinitionBase {
 
     this.ephemeralStorageGiB = props.ephemeralStorageGiB;
 
+    // validate the cpu and memory size for the Windows operation system family.
+    if (props.runtimePlatform?.operatingSystemFamily?._operatingSystemFamily.includes('WINDOWS')) {
+      this.checkFargateWindowsBasedTasksSize(props.cpu!, props.memoryMiB!, props.runtimePlatform!);
```

Comment:
Can you add a comment similar to this? 
```suggestion
      // We know that props.cpu and props.memoryMiB are defined because an error would have been thrown previously if they were not. But, typescript is not able to figure this out, so using the ! operator here to let the type-checker know they are defined. 
      this.checkFargateWindowsBasedTasksSize(props.cpu!, props.memoryMiB!, props.runtimePlatform!);
```